util/fipstools/acvp/modulewrapper/modulewrapper.cc

```diff
@@ -1376,6 +1378,64 @@ static bool GetConfig(const Span<const uint8_t> args[],
         "curve": ["ED-25519"],
         "pure": true,
         "preHash": false
+      },)"
+  R"({
+        "algorithm": "ML-DSA",
+        "mode": "keyGen",
+        "revision": "FIPS204",
+        "parameterSets": ["ML-DSA-44", "ML-DSA-65", "ML-DSA-87"]
+      },{
+        "algorithm": "ML-DSA",
+        "mode": "sigGen",
+        "revision": "FIPS204",
+        "capabilities": [
+        {
+          "parameterSets": [
+            "ML-DSA-44",
+            "ML-DSA-65",
+            "ML-DSA-87"
+          ],
+          "messageLength": [
+            {
+              "min": 8,
+              "max": 65536,
```

Comment:
Why this min/max? Does this affect what is tested or what is approved?

Reply:
FIPS 204 defines the context string as a" byte string of 255 or fewer bytes which can be empty (default)".

Reply:
Should this test min 0, max 255?

Reply:
Reading through the document again, I see now that `messageLength` is the length of the message to be signed. `contextLength` is the length of the context string.

Looking at Table 7 https://pages.nist.gov/ACVP/draft-celi-acvp-ml-dsa.html#name-ml-dsa-siggen-capability-pr, which is where this capabilities string comes from, you'll see that message length's valid values are:

`Min: 8, Max: 65536, Incr: 8`

as defined in my capabilities JSON and `contextLength` is `Min: 0, Max: 2040, Incr: 8`. We're not testing context signing, which is why that capability isn't included.